### PR TITLE
Add unit tests with github actions

### PR DESCRIPTION
**Pull request by nh13**
Note: the following secret environment variables were added:
- `SONATYPE_USER`
- `SONATYPE_PASS`
- `PGP_PASSPHRASE`
- `PGP_SECRET` (from `gpg --armor --export-secret-keys $LONG_ID | base64 | pbcopy`)

### CODE REVIEW

File: .github/workflows/unittests.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        with:
+          java-version: openjdk@8
+        env:
+          ACTIONS_ALLOW_UNSECURE_COMMANDS: true
+      - name: Unit Tests
+        run: |
+          sbt +clean coverage +test && sbt coverageReport coverageAggregate
+          find $HOME/.sbt -name "*.lock" | xargs rm
+          find $HOME/.ivy2 -name "ivydata-*.properties" | xargs rm
+          #bash <(curl -s https://codecov.io/bash) -c -F unittests -f '!*.txt'
```
Comment (jacarey) [inline .github/workflows/unittests.yaml]:
```suggestion
```

File: .github/workflows/unittests.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Setup Scala
+        uses: olafurpg/setup-scala@v1
+        with:
+          java-version: openjdk@8
+        env:
+          ACTIONS_ALLOW_UNSECURE_COMMANDS: true
```
Comment (jacarey) [inline .github/workflows/unittests.yaml]: Is these needed with more recent versions of https://github.com/olafurpg/setup-scala (it is on v12 now and I didn't see it referenced in docs)

File: .github/workflows/unittests.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Setup Scala
+        uses: olafurpg/setup-scala@v1
+        with:
+          java-version: openjdk@8
+        env:
+          ACTIONS_ALLOW_UNSECURE_COMMANDS: true
```
Comment (jacarey) [inline .github/workflows/unittests.yaml]: Same as above.

File: .github/workflows/unittests.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+          flags: unittests # optional
+          fail_ci_if_error: true # optional (default = false)
+          verbose: true # optional (default = false)
+      - name: Cache .ivy2
+        uses: actions/cache@v1
+        with:
+          path: ~/.ivy2
+          key: ${{ runner.os }}-ivy2-${{ hashFiles('**/build.sbt') }}
+      - name: Cache .sbt
+        uses: actions/cache@v1
```
Comment (jacarey) [inline .github/workflows/unittests.yaml]: actions/cache is up to v2 now

File: .github/workflows/unittests.yaml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+          find $HOME/.ivy2 -name "ivydata-*.properties" | xargs rm
+          #bash <(curl -s https://codecov.io/bash) -c -F unittests -f '!*.txt'
+      - name: Code Coverage
+        uses: codecov/codecov-action@v1
+        with:
+          flags: unittests # optional
+          fail_ci_if_error: true # optional (default = false)
+          verbose: true # optional (default = false)
+      - name: Cache .ivy2
+        uses: actions/cache@v1
```
Comment (jacarey) [inline .github/workflows/unittests.yaml]: update to v2?

**Review by jacarey — APPROVED** [review]
Some revision updates and comment removal. Otherwise LGTM

File: .github/workflows/unittests.yaml [hunk 6]
```diff
@@ -0,0 +1,64 @@
+name: unit tests
+
+on:
+  push:
+  pull_request:
```
Comment (sstadick) [inline .github/workflows/unittests.yaml]:
```suggestion
on: [push, pull_request]
```

**Review by sstadick — APPROVED** [review]

File: .github/workflows/unittests.yaml (hunk 2 above)
Comment (nh13) [inline .github/workflows/unittests.yaml]:
```suggestion
      - name: Setup Scala
        uses: olafurpg/setup-scala@v12
        with:
          java-version: openjdk@8
```

File: .github/workflows/unittests.yaml (hunk 5 above)
Comment (nh13) [inline .github/workflows/unittests.yaml]:
```suggestion
        uses: actions/cache@v2
```

File: .github/workflows/unittests.yaml (hunk 4 above)
Comment (nh13) [inline .github/workflows/unittests.yaml]:
```suggestion
        uses: actions/cache@v2
```

File: .github/workflows/unittests.yaml (hunk 3 above)
Comment (nh13) [inline .github/workflows/unittests.yaml]:
```suggestion
      - name: Setup Scala
        uses: olafurpg/setup-scala@v12
        with:
          java-version: openjdk@8
```

File: .github/workflows/unittests.yaml [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Setup Scala
+        uses: olafurpg/setup-scala@v12
+        with:
+          java-version: openjdk@8
```
Comment (nh13) [inline .github/workflows/unittests.yaml]:
```suggestion
          java-version: openjdk@1.9.0
```

File: .github/workflows/unittests.yaml [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+      SONATYPE_USER: ${{ secrets.SONATYPE_USER }}
+      SONATYPE_PASS: ${{ secrets.SONATYPE_PASS }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Setup Scala
+        uses: olafurpg/setup-scala@v12
+        with:
+          java-version: openjdk@8
```
Comment (nh13) [inline .github/workflows/unittests.yaml]:
```suggestion
          java-version: openjdk@1.9.0
```

**Comment by nh13** [comment]
Note: added secrets at the organization level, and scoped them for `sopt`, `commons`, `fgbio`, and `dagr`

**Pull request merged**